modules/dashboard/src/components/Admin/Admin.js

```diff
@@ -1,14 +1,122 @@
+    ethProvider,
+    isLegacy = false,
+    toxicBytecode,
+  ) => {
+    const proxyFactory = new Contract(proxyFactoryAddress, ProxyFactory.abi, ethProvider);
+    // Calculates xpub -> address without the last "/<index>" part of the path
+    const xkeysToSortedKthAddresses = (xkeys) =>
+      xkeys
+        .map((xkey) =>
+          isLegacy
+            ? fromExtendedKey(xkey).address
+            : fromExtendedKey(xkey).derivePath("0").address
+        )
+        .sort((a, b) => (parseInt(a, 16) < parseInt(b, 16) ? -1 : 1));
+    const ownerAddresses = xkeysToSortedKthAddresses(owners);
+    // console.log(`Got ownerAddresses: ${JSON.stringify(ownerAddresses)}`);
+    const proxyBytecode = toxicBytecode || await proxyFactory.functions.proxyCreationCode();
+    return getAddress(
+      solidityKeccak256(
+        ["bytes1", "address", "uint256", "bytes32"],
+        [
+          "0xff",
+          proxyFactoryAddress,
+          solidityKeccak256(
+            ["bytes32", "uint256"],
+            [
+              keccak256(
+                // see encoding notes
+                new Interface(MinimumViableMultisig.abi).functions.setup.encode([
+                  ownerAddresses,
+                ]),
+              ),
+              0,
+            ],
+          ),
+          solidityKeccak256(
+            ["bytes", "uint256"],
+            [`0x${proxyBytecode.replace("0x", "")}`, minimumViableMultisigAddress],
+          ),
+        ],
+      ).slice(-40),
+    );
+  }
+
+  const scanForFactory = async (owners, expectedMultisig) => {
+    console.log(`Scanning for expected multisig: ${expectedMultisig}`);
+    if (!owners || !expectedMultisig) return;
+    const provider = getDefaultProvider("homestead");
+    for (const multisig of historicalData.MinimumViableMultisigAddresses) {
+      for (const factory of historicalData.ProxyFactoryAddresses) {
+        for (const bytecode of historicalData.proxyBytecode) {
+          for (const legacy of [true, false]) {
+            const bc = bytecode.substring(0,8) || `0x000000`;
+            let calculated = await legacyGetCreate2MultisigAddress(
+              owners,
+              factory,
+              multisig,
+              provider,
+              legacy,
+              bytecode
+            );
+            console.log(
+              `factory ${factory} + multisig ${multisig} + bytecode ${bc}... + legacy ${legacy}` +
+              ` => ${calculated}`,
+            );
+            if (calculated === expectedMultisig) {
+              console.log(`MATCH DETECTED`);
+              console.log(`calculated: ${calculated}`);
+              console.log(`legacy:     ${legacy}`);
+              console.log(`multisig:   ${multisig}`);
+              console.log(`factory:    ${factory}`);
+              console.log(`bytecode:   ${bytecode}`);
+              return [legacy, multisig, factory, bytecode];
+            }
+          }
+        }
+      }
+    }
+    return "oh no none match :("
+  }
```

Comment:
why not use the same functions/logic as is in the hub? those can easily be converted into admin endpoints without having to recreate all the logic here. Check out the last couple of purpose built functions [here](https://github.com/ConnextProject/indra/blob/staging/modules/node/src/admin/admin.service.ts#L267)